spec/unit/shoulda/matchers/active_record/have_db_column_matcher_spec.rb

```diff
@@ -98,6 +98,18 @@
     end
   end
 
+  context 'with array option' do
+    it 'accepts a column of correct array' do
+      expect(with_table(:phones, :string, array: false)).
+        to have_db_column(:phones).with_options(array: false)
+    end
+
+    it 'rejects a column of wrong array' do
+      expect(with_table(:phones, :string, array: true)).
+        not_to have_db_column(:phones).with_options(array: false)
+    end
+  end
+
```

Comment:
Good evening. I also thought about doing this, but the test will never successfully in unsupported database.